react/src/lib/components/SubsurfaceViewer/layers/labeled_points/labeledPointsLayer.stories.tsx

```diff
@@ -0,0 +1,141 @@
+import React from "react";
+import { ComponentStory, ComponentMeta } from "@storybook/react";
+import { create, all } from "mathjs";
+
+import SubsurfaceViewer from "../../SubsurfaceViewer";
+import { default as LabeledPointsLayer } from "./labeledPointsLayer";
+import { default as AxesLayer } from "../axes/axesLayer";
+
+export default {
+    component: SubsurfaceViewer,
+    title: "SubsurfaceViewer / Experimental LabeledPoints Layer",
+} as ComponentMeta<typeof SubsurfaceViewer>;
+
+const defaultParameters = {
+    docs: {
+        inlineStories: false,
+        iframeHeight: 500,
+    },
+};
+
+// Small example using labeledPointsLayer.
+const labeledPointsLayer = new LabeledPointsLayer({
+    id: "labeledPoints-layer",
+    /*eslint-disable */
+    pointsData:   [  0,  0,  5,  // Vertex 1, x, y, z
+                    10,  0,  5,  // Vertex 2, x, y, z
+                    10, 10,  5,  // ...
+                     0, 10,  0,
+                     5, -5, 10,
+                    11, -4,  6,
+                    11,  0,  7,
+                    17,  0,  8
+                    ],    
+    /*eslint-enable */
+    color: [255, 100, 100],
+    pointRadius: 10,
+    radiusUnits: "pixels",
+    ZIncreasingDownwards: true,
+});
+
+const smallAxesLayer = new AxesLayer({
+    id: "axes_small",
+    bounds: [-10, -10, 0, 20, 10, 10],
+});
+
+export const SmallLabeledPointsLayer: ComponentStory<
+    typeof SubsurfaceViewer
+> = (args) => {
+    return <SubsurfaceViewer {...args} />;
+};
+
+SmallLabeledPointsLayer.args = {
+    id: "map",
+    layers: [smallAxesLayer, labeledPointsLayer],
+    bounds: [-10, -10, 17, 10],
+    views: {
+        layout: [1, 1],
+        viewports: [
+            {
+                id: "view_1",
+                show3D: true,
+            },
+        ],
+    },
+};
+
+SmallLabeledPointsLayer.parameters = {
+    docs: {
+        ...defaultParameters.docs,
+        description: {
+            story: "Point coordinates are given as native javascript array.",
+        },
+    },
+};
+
+// Huge example using labeledPointsLayer.
+const sideSize = 10000;
+const pointsCount = 100000;
+
+const math = create(all, { randomSeed: "1234" });
+
+type TRandomNumberFunc = () => number;
+
+const randomFunc = ((): TRandomNumberFunc => {
+    if (math?.random) {
+        return () => {
+            const val = math.random?.(sideSize);
+            return val ? val : 0.0;
+        };
+    }
+    return () => Math.random() * sideSize;
+})();
+
+const hugePointsLayer = new LabeledPointsLayer({
+    id: "labeledPoints-layer",
+    pointsData: Array(pointsCount * 3)
+        .fill(0)
```

Comment:
Do you really need the fill here ?

Reply:
fill() is needed because Array constructor creates a sparse array of the length provided but not containing any data. It isn't even filled with undefined values. That is map() wont iterate along the array.

Reply:
OK, thank you